pkg/handlers/generic/mutation/encryption/inject_test.go

```diff
@@ -0,0 +1,233 @@
+	patchGenerator := func() mutation.GeneratePatches {
+		client, err := helpers.TestEnv.GetK8sClientWithScheme(clientScheme)
+		Expect(err).To(BeNil())
+		return mutation.NewMetaGeneratePatchesHandler(
+			"",
+			client,
+			NewPatch(client, testTokenGenerator)).(mutation.GeneratePatches)
+	}
+
+	encryptionVar := []runtimehooksv1.Variable{
+		capitest.VariableWithValue(
+			clusterconfig.MetaVariableName,
+			carenv1.Encryption{
+				Providers: &carenv1.EncryptionProviders{
+					AESCBC:    &carenv1.AESConfiguration{},
+					Secretbox: &carenv1.SecretboxConfiguration{},
+				},
+			},
+			VariableName,
+		),
+	}
+	encryptionMatchers := []capitest.JSONPatchMatcher{
+		{
+			Operation: "add",
+			Path:      "/spec/template/spec/kubeadmConfigSpec/files",
+			ValueMatcher: ContainElements(
+				SatisfyAll(
+					HaveKeyWithValue(
+						"path", "/etc/kubernetes/encryptionconfig.yaml",
+					),
+					HaveKeyWithValue(
+						"permissions", "0600",
+					),
+					HaveKeyWithValue(
+						"contentFrom",
+						map[string]interface{}{
+							"secret": map[string]interface{}{
+								"key":  "config",
+								"name": defaultEncryptionSecretName(request.ClusterName),
+							},
+						},
+					),
+				),
+			),
+		},
+		{
+			Operation: "add",
+			Path:      "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration",
+			ValueMatcher: HaveKeyWithValue(
+				"apiServer",
+				HaveKeyWithValue(
+					"extraArgs",
+					map[string]interface{}{
+						"encryption-provider-config": "/etc/kubernetes/encryptionconfig.yaml",
+					},
+				),
+			),
+		},
+	}
+
+	// Create cluster before each test
+	BeforeEach(func(ctx SpecContext) {
+		client, err := helpers.TestEnv.GetK8sClientWithScheme(clientScheme)
+		Expect(err).To(BeNil())
+
+		Expect(client.Create(
+			ctx,
+			&clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      request.ClusterName,
+					Namespace: metav1.NamespaceDefault,
+				},
+			},
+		)).To(BeNil())
+	})
+
+	// Delete cluster after each test
+	AfterEach(func(ctx SpecContext) {
+		client, err := helpers.TestEnv.GetK8sClientWithScheme(clientScheme)
+		Expect(err).To(BeNil())
+
+		Expect(client.Delete(
+			ctx,
+			&clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      request.ClusterName,
+					Namespace: metav1.NamespaceDefault,
+				},
+			},
+		)).To(BeNil())
+	})
+
+	// Test that encryption secret patch is skipped if it is already applied.
+	// noOpEncryptionConfigDef := capitest.PatchTestDef{
+	// 	Name:        "skip patching encryption config if default encryption config secret exists",
+	// 	Vars:        encryptionVar,
+	// 	RequestItem: request.NewKubeadmControlPlaneTemplateRequestItem(""),
+	// 	ExpectedPatchMatchers: []capitest.JSONPatchMatcher{
+	// 		{
+	// 			Operation:    "",
+	// 			Path:         "",
+	// 			ValueMatcher: Equal(nil),
+	// 		},
+	// 	},
+	// 	UnexpectedPatchMatchers: encryptionMatchers,
+	// }
+
+	// Context("Default encryption provider secret already exists", func() {
```

Comment:
I was not able to get existing test helper to work with use case where the Response contains empty patch.
```
 <[]v1alpha1.GeneratePatchesResponseItem | len:1, cap:1>: [
          {
              UID: "4fa296dd-abbb-4629-9afd-c8457890248f",
              PatchType: "JSONPatch",
              Patch: "[]",
          },
      ]
```